src/radio.cpp

```diff
@@ -100,18 +100,18 @@ void zmq::radio_t::xwrite_activated (pipe_t *pipe_)
 
 void zmq::radio_t::xpipe_terminated (pipe_t *pipe_)
 {
-    for (subscriptions_t::iterator it = subscriptions.begin (); it != subscriptions.end (); ) {
```

Comment:
There's a reason I did this change, perhaps there should be a comment.
Iterators are invalidated when deleted, so this will cause a read of freed memory and potentially a crash.

See https://github.com/zeromq/libzmq/pull/1771